src/main/java/com/epam/team1/emotional/help/service/AvatarService.java

```diff
@@ -0,0 +1,105 @@
+package com.epam.team1.emotional.help.service;
+
+import com.epam.team1.emotional.help.dto.MessageResponse;
+import com.epam.team1.emotional.help.exception.StorageFileNotFoundException;
+import com.epam.team1.emotional.help.model.User;
+import com.epam.team1.emotional.help.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.UrlResource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import org.webjars.NotFoundException;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class AvatarService {
+
+    private final UserService userService;
+
+    private final UserRepository userRepository;
+
+    @Value("${path.basePackagePath}")
+    private String basePackagePath;
+
+
+    String pathSeparator = File.separator;
+
+    public MessageResponse save(@NotNull MultipartFile avatar) {
+        if (avatar.isEmpty()) {
+            throw new RuntimeException("avatar must not be empty");
+        }
+        User currentUser = userService.getCurrentUser().orElseThrow(() -> new NotFoundException("User not found"));
+
+        String avatarOriginalFilename = avatar.getOriginalFilename();
+        File file = new File(basePackagePath + pathSeparator + currentUser.getEmail() + pathSeparator + avatarOriginalFilename);
+
+        boolean wasSuccessful = file.mkdirs();
+        log.info("directories creation were successful = " + wasSuccessful + " it also can be not cussessfull becouse if it already exists");
+        log.info("file absolute = " + file.getAbsoluteFile());
+        log.info(" can read and wright  = " + file.canRead() + "" + file.canWrite());
+        try {
+            avatar.transferTo(file);
+            currentUser.setImage(avatarOriginalFilename);
+            userRepository.save(currentUser);
+        } catch (Exception e) {
+            e.printStackTrace();
+            log.error("exception is " + e);
```

Comment:
Why do you use `printStackTrace` if you use logging? Add more info to the log + you can use placeholders in log messages, for example,  log.error("Some text, exception: {}", e);